### ISSUE TITLE
bench: BenchmarkOrderBy causes crdb to panic with a faulty RowChannel

### ISSUE THREAD

**Issue by knz** [cockroachdb/cockroach#21289]
`make bench PKG=./pkg/bench TESTFLAGS='-count 1 -benchmem' BENCHTIMEOUT=1h` eventually fails with:

```
F180106 17:32:32.438831 1554035 sql/distsqlrun/base.go:405  RowChannel already done or closed: 1
goroutine 1554035 [running]:
github.com/cockroachdb/cockroach/pkg/util/log.getStacks(0x2f63700, 0x150748e6077de7e0, 0x2a09edf, 0x16)
        /data/home/kena/src/go/src/github.com/cockroachdb/cockroach/pkg/util/log/clog.go:871 +0xcf
github.com/cockroachdb/cockroach/pkg/util/log.(*loggingT).outputLogEntry(0x2f63700, 0xc400000004, 0x2a09edf, 0x16, 0x195, 0xc42342c300, 0x24)
        /data/home/kena/src/go/src/github.com/cockroachdb/cockroach/pkg/util/log/clog.go:755 +0x5c1
github.com/cockroachdb/cockroach/pkg/util/log.addStructured(0x2d429e0, 0xc4200161d8, 0x4, 0x2, 0x1e8ec82, 0x25, 0xc42735fb10, 0x1, 0x1)
        /data/home/kena/src/go/src/github.com/cockroachdb/cockroach/pkg/util/log/structured.go:152 +0x308
github.com/cockroachdb/cockroach/pkg/util/log.logDepth(0x2d429e0, 0xc4200161d8, 0x1, 0xc400000004, 0x1e8ec82, 0x25, 0xc42735fb10, 0x1, 0x1)
        /data/home/kena/src/go/src/github.com/cockroachdb/cockroach/pkg/util/log/log.go:55 +0x8c
github.com/cockroachdb/cockroach/pkg/util/log.Fatalf(0x2d429e0, 0xc4200161d8, 0x1e8ec82, 0x25, 0xc42735fb10, 0x1, 0x1)
        /data/home/kena/src/go/src/github.com/cockroachdb/cockroach/pkg/util/log/log.go:149 +0x7e
github.com/cockroachdb/cockroach/pkg/sql/distsqlrun.(*RowChannel).ConsumerDone(0xc426c44c90)
        /data/home/kena/src/go/src/github.com/cockroachdb/cockroach/pkg/sql/distsqlrun/base.go:405 +0xda
github.com/cockroachdb/cockroach/pkg/sql/distsqlrun.(*distinct).Next(0xc4207d2a00, 0xc4207d2be0, 0x1, 0x1, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
        /data/home/kena/src/go/src/github.com/cockroachdb/cockroach/pkg/sql/distsqlrun/distinct.go:209 +0x489
github.com/cockroachdb/cockroach/pkg/sql/distsqlrun.Run(0x2d429a0, 0xc42b4b6200, 0x2d433a0, 0xc4207d2a00, 0x2d2f4a0, 0xc42a1a4240)
        /data/home/kena/src/go/src/github.com/cockroachdb/cockroach/pkg/sql/distsqlrun/base.go:153 +0x74
github.com/cockroachdb/cockroach/pkg/sql/distsqlrun.(*distinct).Run(0xc4207d2a00, 0xc420b347e8)
        /data/home/kena/src/go/src/github.com/cockroachdb/cockroach/pkg/sql/distsqlrun/distinct.go:72 +0x6b
created by github.com/cockroachdb/cockroach/pkg/sql/distsqlrun.(*Flow).Start
        /data/home/kena/src/go/src/github.com/cockroachdb/cockroach/pkg/sql/distsqlrun/flow.go:419 +0x3aa
exit status 255
```

cc @RaduBerinde @asubiotto

**Comment by petermattis** [comment]
Probably my fault. Also probably fixed by #21282.

**Comment by petermattis** [comment]
Definitely my fault. Definitely fixed by #21282.

**Comment by petermattis** [comment]
This was fixed by #21282.
